source/math/IntRandomGenerator.ooc

```diff
@@ -35,35 +59,31 @@ IntRandomGenerator: class {
 	}
 }
 
-IntUniformRandomGenerator: class extends IntRandomGenerator {
-	init: func {
-		super(FloatUniformRandomGenerator new(0, 100))
-	}
-	init: func ~withSeed (seed: Int) {
-		super(FloatUniformRandomGenerator new(0, 100, seed))
-	}
-	init: func ~withParameters (min, max: Int, seed := Time microtime()) {
-		super(FloatUniformRandomGenerator new(min, max, seed))
-	}
-}
-
-IntGaussianRandomGenerator: class extends IntRandomGenerator {
+IntGaussianRandomGenerator: class {
+	_backend: FloatRandomGenerator
 	init: func {
-		super(FloatGaussianRandomGenerator new(0.0f, 10.0f))
+		this _backend = FloatGaussianRandomGenerator new(0.0f, 1.0f)
 	}
 	init: func ~withSeed (seed: Int) {
-		super(FloatGaussianRandomGenerator new(0.0f, 10.0f, seed))
+		this _backend = FloatGaussianRandomGenerator new(0.0f, 1.0f, seed)
 	}
 	init: func ~withParameters (mu, sigma: Float, seed := Time microtime()) {
-		super(FloatGaussianRandomGenerator new(mu, sigma, seed))
+		this _backend = FloatGaussianRandomGenerator new(mu, sigma, seed)
 	}
-	init: func ~withBackend (backend: FloatGaussianRandomGenerator) {
-		super(backend)
+	init: func ~withUniformBackend (=_backend)
+	init: func ~withUniformBackendAndParameters (mu, sigma: Float, backend: FloatUniformRandomGenerator) {
+		this _backend = FloatGaussianRandomGenerator new~withBackendAndParameters(mu, sigma, backend)
 	}
-	init: func ~withUniformBackend (backend: FloatUniformRandomGenerator) {
-		super(FloatGaussianRandomGenerator new~withBackendAndParameters(0.0f, 10.0f, backend))
+	free: func {
+		this _backend free()
```

Comment:
Do I always call super() even if there is no inheritance? What about IntUniformRandomGenerator, do I need a free() which justs calls super()?